Fix for issue #111

drajarshi opened the pull request:
Fix for issue #111 
The changes allow:
1. to read the stat names as is without removing the regex symbols.
2. Customize the stat names without changing the regex used to fetch their corresponding values.

drajarshi committed:
Remove regex symbols from get_stats script output

Modify example_stats.yml to include stat names separately and
get_stats.py to parse the modified yaml.

Signed-off-by: Rajarshi Das <drajarsh@gmail.com>

drajarshi commented:
I clicked on the 'Details' in the build failure link above and saw a message to provide access to my github account as below:
![accelsim-pull_request_112-failure-authorize_page2](https://user-images.githubusercontent.com/22958960/161661408-d177b3fc-a53a-4900-8fb3-5d531e5a2b6e.png)

I provided the permissions. Subsequently, I got to a second page indicating 'permission denied' as below:
![accelsim-pull_request_112-failure-jenkins-dashboard-error-msg-page3](https://user-images.githubusercontent.com/22958960/161661522-51a6b232-7fb8-470d-abe4-14ac94560ca1.png)

Please suggest how to proceed further in order to fix the build issue.

drajarshi commented:
I clicked on the 'Details' in the jenkins build failure link and saw a message to provide access to my github account as below:
![accelsim-pull_request_112-failure-authorize_page2](https://user-images.githubusercontent.com/22958960/161661408-d177b3fc-a53a-4900-8fb3-5d531e5a2b6e.png)

I provided the permissions. Subsequently, I got to a second page indicating 'permission denied' as below:
![accelsim-pull_request_112-failure-jenkins-dashboard-error-msg-page3](https://user-images.githubusercontent.com/22958960/161661522-51a6b232-7fb8-470d-abe4-14ac94560ca1.png)

Please suggest how to proceed further in order to fix the build issue.

JRPan commented:
Hi, 

The build was finished. Unfortunately, the Jenkins log is not viewable public. You may ignore the failed message. You don't have to worry about it. We'll assign someone for a review and provide any feedback if needed.

Thanks

drajarshi commented:
Thank you @JRPan

drajarshi committed:
Remove regex symbols from correl_mappings.py

Modify correl_mappings.py by removing regex symbols so that
plot-correlation.py can use the csv output from the fixed
get_stats.py script.

Signed-off-by: Rajarshi Das <drajarsh@gmail.com>

drajarshi commented:
@JRPan 
The modifications to get_stats.py and example_stats.yml generate a csv output which does not contain regex.
Modified correl_mappings.py so that the plot-correlation.py script can use the modified csv to generate correlation data as before.

This change also allows completely removing the regular expressions from the correl_mappings.py file making it easier to read the simulation specific stat names. 
It also allows using a modified stat name (which may differ from what is actually seen in the simulation output) in the correl_mappings.py. 
The only requirement is that the modified stat name should match the stat name (the key for each stat name mapping) that has been used in the example_stats.yml (and used to run get_stats.py).

Thanks

JRPan commented:
Hi sorry, we've been busy. 

Looks like the Jenkins is still failing. Are you sure the correl_mappings.py is working? There were no plots generated by Jenkins. This could happen when formatting changes and the python script was not able to find the regex string.

drajarshi commented:
@JRPan 

Thank you for your comment.
I tested the flow once again with my committed version of get_stats.py, and example_stats.yml.
For correl_mappings.py, I used the committed version + changes for my GPU device (since my GPU device is not listed in the existing config_maps).
I am able to see that the plot-correlator.py is generating plots. Please see the detailed output of my test in the issue #111 
I am unable to test the committed version of correl_mappings.py directly since I do not have the hardware run data for any of the existing devices in the config_maps.
So, in order to debug further, 
it would be helpful if you can share sample hardware run data (output of run_hw.py) for any of the devices listed in config_maps, and also the job log of jenkins, so that I can test the exact committed version of correl_mappings.py, and recreate the problem.

Thanks.

drajarshi commented:
@JRPan 
The other thing to note is that after the changes, correl-mappings.py does not contain any regex expressions. This is because get_stats.py now generates a output .csv only with the stat names. So, plot-correlator.py can now use only the stat names in order to correlate the data.

Hence, if the jenkins job is explicitly looking for regex, it will not find any.
Maybe the job needs to be modified to look for stat names now instead of regex? Just a thought.

Thanks.

drajarshi commented:
@JRPan 
Can I set up a call to discuss this? I would like to understand more on why jenkins fails and how I can fix the issue. It is not obvious to me at this time.

Thanks

JRPan committed:
Merge branch 'dev' into drajarshi-dev-get_stats-script
